course/assignments/R-intro-lectures-assignments.md

```diff
@@ -0,0 +1,60 @@
+# Week 1 (4 marks) Intro to programming
+1. Why is it beneficial to use a programming language rather than a spreadsheet software for data analyses?
+2. Get setup with Rstudio at home (or a lab computer after hours). Install R, Rstudio, and the necessary packages (tidyverse, rmarkdown). Use `install.packages(<package_name>)`. Load all the libraries you installed into your environment with `library(<package_name>)` and send us the output of running `sessionInfo()`.
+3. What is Rmarkdown and why are we using it in this class? How do you execute a chunk of code in an Rmarkdown document?
+4. Is there anything in particular you are excited about learning about theoretical ecological population models and about using R for data analyses and modelling?
+
+# Week 2 (8 marks) Rmarkdown, assignment, functions, vectors, data frames
+All the following should be done with Rmarkdown.
+
+1. Assignment
+    a. Assign the value `5` to the variable/object `a`.
+    b. Assign the result of `10/3` to the variable `b`.
+    c. Assign the product of `a` and `b` to `c`.
+    d. Write a function that multiplies two numbers and returns the result. Use it to assign the product of `a` and `b` to `c`.
+2. Vectors
+    a. Create a vector `v` with all integers 1-30.
+    b. Use `v` to create a new vector `v_square` with the square of element 3, 6, 7, 10, 15, 22, 23, 24, and 30 from `v`. Do _not_ use a for loop.
+    c. Extract every 5th element from `v`, into a new vector `v_every_5th`.
+4. Boolean indexing
+    a. Create a boolean vector `v_bool` of the same length as `v`, indicating which vector elements are bigger than 20.
+    b. Use `v_bool` as an index to extract the elements from `v` that are bigger than 20 and save as `v_over_20`.
+    c. What's the mean of `v_over_20`?
+5. Data frames
+    a. What are the column names of inbuilt dataframe `iris`?
+    b. How can you view the first 5 rows of this data frame?
+    c. How many observations are there?
+    d. Which factor levels are there? What is a factor level and when can it be useful?
+    e. What is the mean value for the sepal.height column?
+6. Save the data frame with only the species and sepal.width columns as csv-file. Load this csv file as a new data frame and run display its summary statistics.
+
+
+# Week 3 (8 marks) dplyr, ggplot
```

Comment:
This will probably be a bit hard as an early assignment, especially for only 8 marks and given that it requires some conceptual thinking about data and wrangling it that the students might not have yet.